ONSdigital/dp-static-file-publisher
# Feature/decrypt and copy image

### What

- Implemented event handler for image published event:
  - Schema defines SrcPath and DstPath, corresponding to a single file that needs to be decrypted and copied to the static bucket.
  - Vault path for file assumed to be: `<vaultPath>/<srcPath>`, where vaultPath is defined by config, and srcPath is obtained from the kafka message.

- Provided kafka producer for testing under cmd/producer/main.go

- Replaced S3Public client with uploader, which is an extension of the s3 client for uploading images using s3manager. (see dp-s3 repository, `uploader.go` for more info)

### How to review

- Make sure code changes make sense
- Make sure unit tests pass
- Try to run the kafka producer for testing: `go run cmd/producer/main.go`
- Optionally, you can check the kafka producer in image api: https://github.com/ONSdigital/dp-image-api/pull/25

### Who can review

Anyone
